Add docs for `fullClone` federalist.json config option (#452)
18F/federalist.18f.gov: pages/documentation/federalist-json.md

sidenav: documentation
---

Federalist-specific configuration can be done in a `federalist.json` file in the root of your project. Currently, we only support the following keys:
- `headers`
- `fullClone`

## `headers`
See [specifying custom headers](/documentation/custom-headers) for details.

## `fullClone`

| values | default? | effect |
| ------ |:--------:| ------ |
| not specified | **Y** | shallow clone |
| `true` | | full clone |
| anything else | | shallow clone |

> By default, Federalist does a "shallow" clone of a single branch of your repository to minimize the space required for and duration of your build. This means that entire git revision is NOT available during Federalist builds which may differ from your local environment. This can cause an issue with some site engines or plugins that generate "last modified" datetimes base on the git revision history for particular files.

Reviewer: ```suggestion
By default, Federalist does a "shallow" clone of a single branch of your repository to minimize the space required for the duration of your build. This means that entire git revision is NOT available during Federalist builds which may differ from your local environment. This can cause an issue with some site engines or plugins that generate "last modified" datetimes base on the git revision history for particular files.
```

Author: hmmm, i'm trying to say there are 2 reasons we do the shallow clone:
1. minimize the space
2. minimize the build time

I feel like this says something different if this change is made... But maybe I need to reword this.

maybe
```
By default, Federalist does a "shallow" clone of a single branch of your repository to minimize the required disk space and total time of your build. This means that the entire git revision history is NOT available during Federalist builds which may differ from your local environment. This can cause an issue with some site engines or plugins that generate "last modified" datetimes based on the git revision history for particular files.
```
?

Reviewer: i see .. .there's a bit of a pause there

> By default, Federalist does a "shallow" clone of a single branch of your repository to minimize the space required for and duration of your build. This means that entire git revision is NOT available during Federalist builds which may differ from your local environment. This can cause an issue with some site engines or plugins that generate "last modified" datetimes base on the git revision history for particular files.

Reviewer: ```suggestion
By default, Federalist does a "shallow" clone of a single branch of your repository to minimize the duration of your build as well as the space it requires. This means that entire git revision is NOT available during Federalist builds which may differ from your local environment. This can cause an issue with some site engines or plugins that generate "last modified" datetimes base on the git revision history for particular files.
```

> By default, Federalist does a "shallow" clone of a single branch of your repository to minimize the duration of your build as well as the space it requires. This means that entire git revision is NOT available during Federalist builds which may differ from your local environment. This can cause an issue with some site engines or plugins that generate "last modified" datetimes base on the git revision history for particular files.

Author: ```suggestion
By default, Federalist does a "shallow" clone of a single branch of your repository to minimize the duration of your build as well as the space it requires. This means that the entire git revision history is NOT available during Federalist builds, which may differ from your local environment. This can cause an issue with some site engines or plugins that generate "last modified" datetimes based on the git revision history for particular files.
```
